etc/notes/CHANGES_5.0.0.md

```diff
@@ -18,6 +18,84 @@ The following is a detailed collection of the changes in the major v5 release of
 
 ### Optional callback support migrated to `mongodb-legacy`
 
+Node v5 drops support for callbacks in favor of a pure async-await API.  Below are some strategies for
+callback users to adopt driver v5 in recommendation order.
+
+The Node driver team understands that a callback to promise migration can be a non-trivial refactor.  To help inform your migration strategy, we've outlined three different approaches below.
+
+#### Migrate to promise based api (recommended!)
+
+The Node team strongly encourages anyone who is able to migrate from callbacks to promises to do so.  Adopting the
+regular driver API will streamline the adoption of future driver updates, as well as provide better Typescript support
+than the other options outlined in this document.
+
+The promise-based API is identical to the callback API except:
+
+- no callback is accepted as the last argument
+- a promise is always returned
+
+For example, with a findOne query:
+
+```typescript
+// callback-based API
+collection.findOne({ name: 'john snow'}, (error, result) => {
+  if (error) {
+    doSomethingWithError(error);
+    return;
+  }
+
+  doSomethingWithResult(result);
+})
+
+// promise-based API
+collection.findOne({ name: 'john snow'}).then(
+  (result) => doSomethingWithResult(result),
+  (error) => doSomethingWithError(error)
+);
```

Comment:
Do we want to be recommending this manual promise wrapping? If `async` `await` isn't available, `.then`/`.catch` is what _most_ people should use attaching to an existing promise chain (possibly by just returning it: `return c.findOne()`) 

If folks need to inline this code inside of a place that is callback based they should prefer callbackify because of how it correctly does the error handling for promise rejection

Reply:
```suggestion
// promise-based API
collection.findOne({ name: 'john snow'})
  .then((result) => doSomethingWithResult(result))
  .catch((error) => doSomethingWithError(error));
```

is this what you're suggesting?

Reply:
Yep! 

Maybe take it a step further and don't invoke handlers: `doSomethingWithResult` and instead leave comments: `/* Do something with result */`. We know how mixing the two paradigms gets tricky so I don't want to have code samples that suggest a way that could be error prone, but I don't feel very strongly at least if we're demonstrating usage with the typical `.catch` handling.